### ISSUE TITLE
mulgpu training error

### ISSUE THREAD

**Issue by jo-dean** [ultralytics/ultralytics#13547]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

``Hello,I have two 4090 in my machine.and my img_size is 3840*2160,this is my training code：
```
from ultralytics import YOLO, RTDETR

model = RTDETR('rtdetr-x.pt')

# Train the model
dataset_config_path = 'ultralytics/ultralytics/cfg/datasets/kmc.yaml'
results = model.train(data=dataset_config_path, epochs=200, batch=4, imgsz=1280, 
        workers=8, device=[0,1],patience=150)
```
**my input img size is 1280，and my label num is 34,my traindata only labeled 1 class.such as my classes is:person,car,cat....,and my training data only labeled person. when trained ended.the val_batch0_pred.jpg is boxed with other class,like cat,cat,plane... etc.**
**But when i just use one gpu，like 0 or 1.the result is right.and the  result of val_batch0_pred.jpg is right**

this is my dataset config：
```
# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: Dataset/kmc-train # dataset root dir
train: images/train # train images (relative to 'path') 4 images
val: images/val # val images (relative to 'path') 4 images
test: # test images (optional)


# number of classes
nc: 34
# class names
names:[0:xx...]
```
this is my args.yaml
```
task: detect
mode: train
model:rtdetr-x.pt
data: /ultralytics/cfg/datasets/kmc.yaml
epochs: 200
time: null
patience: 150
batch: 8
imgsz: 1280
save: true
save_period: -1
cache: false
device:
- 0
- 1
workers: 8
project: null
name: train6
exist_ok: false
pretrained: true
optimizer: auto
verbose: true
seed: 0
deterministic: true
single_cls: false
rect: false
cos_lr: false
close_mosaic: 10
resume: false
amp: true
fraction: 1.0
profile: false
freeze: null
multi_scale: false
overlap_mask: true
mask_ratio: 4
dropout: 0.0
val: true
split: val
save_json: false
save_hybrid: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
vid_stride: 1
stream_buffer: false
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
embed: null
show: false
save_frames: false
save_txt: false
save_conf: false
save_crop: false
show_labels: true
show_conf: true
show_boxes: true
line_width: null
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: false
opset: null
workspace: 4
nms: false
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
pose: 12.0
kobj: 1.0
label_smoothing: 0.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
bgr: 0.0
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
auto_augment: randaugment
erasing: 0.4
crop_fraction: 1.0
cfg: null
tracker: botsort.yaml
save_dir: runs/detect/train6

```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
Hello @jo-dean,

Thank you for reaching out and providing detailed information about your issue. It looks like you're encountering a problem with multi-GPU training where the validation results are incorrect when using both GPUs, but correct when using a single GPU.

To assist you better, could you please provide a minimal reproducible example? This will help us replicate the issue on our end and investigate it further. You can find guidelines for creating a minimal reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example). Ensuring that we can reproduce the bug is crucial for us to identify and resolve the issue effectively.

Additionally, please verify that you are using the latest versions of `torch` and `ultralytics`. You can upgrade your packages using the following commands:

```bash
pip install --upgrade torch
pip install --upgrade ultralytics
```

Once you've confirmed that you're on the latest versions, please try running your training script again to see if the issue persists.

If the problem continues, please share the minimal reproducible code example along with any additional logs or error messages you encounter. This will greatly aid in diagnosing the problem.

Looking forward to your response!
